src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java

```diff
@@ -533,6 +494,34 @@ public String switchToNewBrowserTab() {
     return driver.getCurrentUrl();
   }
 
+  private String getNewTab(String parentTab) {
+    Optional<String> newTab = driver
+      .getWindowHandles()
+      .stream()
+      .filter(handleName -> !handleName.equals(parentTab))
+      .findFirst();
+    return newTab.orElseThrow(() -> new NotFoundException("New tab not found!"));
+  }
+
+  private String switchToNewTab(String parentTab) {
+    String newTab = getNewTab(parentTab);
+    driver.switchTo().window(newTab);
+    return newTab;
+  }
+
+  private void waitForLinkOpenedInNewTab(WebElement link) {
+    int tabs = driver.getWindowHandles().size();
+    link.click();
+    new WebDriverWait(driver, TIMEOUT_PAGE_REGISTRATION)
+      .until((Predicate<WebDriver>) input -> driver.getWindowHandles().size() > tabs);

```

Comment:
Extract the `driver.getWindowHandles().size()` into another method `getTabNumber`?